packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts

```diff
@@ -43,44 +37,34 @@ test.describe('PieCookieBanner - Component tests', () => {
 
     [{ name: 'action' }, { name: 'body' }].forEach((elementLevel) => {
         const level = elementLevel.name as Level;
-        test(`should emit the correct event and close the cookie banner when "Accept all" is clicked via element ${elementLevel.name}`, async ({ mount }) => {
+        test(`should emit the correct event and close the cookie banner when "Accept all" is clicked via element ${elementLevel.name}`, async () => {
             // Arrange
-            const events : Array<Event> = [];
-
-            await mount(PieCookieBanner, {
-                props: {} as CookieBannerProps,
-
-                on: {
-                    [ON_COOKIE_BANNER_ACCEPT_ALL]: (event: Event) => events.push(event),
-                },
-            });
+            await pieCookieBannerComponent.load();
+            await pieCookieBannerComponent.listenForEvent(ON_COOKIE_BANNER_ACCEPT_ALL);
 
             // Act
             await pieCookieBannerComponent.clickAcceptAll(level);
+
+            const events = await pieCookieBannerComponent.getCapturedEvents();
             const isCookieBannerVisible = await pieCookieBannerComponent.isCookieBannerVisible();
 
             // Assert
             expect.soft(isCookieBannerVisible).toBe(false);
-            expect(events).toHaveLength(1);
+            expect(events).toContain(ON_COOKIE_BANNER_ACCEPT_ALL);
```

Comment:
```suggestion
            expect(events).toHaveLength(1);
            expect(events).toContain(ON_COOKIE_BANNER_ACCEPT_ALL);
```

Reply:
Or actually, as that's the only event we're listening for, we can probably get away with just checking the length as we were before